### PR TITLE
[Fix #1633] Add MultilineMethodCallIndentation

### LINKED ISSUE #1633: Align the operands of an expression in an assignment spanning multiple lines bug (or me not understanding the config)
I'm running: `0.28.0 (using Parser 2.2.0.2, running on ruby 2.0.0 x86_64-darwin14.0.0)`

With the code:
```ruby
users = User.where(email: 'user@example.com').
             update_all(user_parameters)
```

I get the rubocop warning:

>Align the operands of an expression in an assignment spanning multiple lines

It passes with this which I feel isn't correct indentation:

```ruby
users = User.where(email: 'user@example.com').
        update(user_parameters)
```

I'm not sure if this is a bug or my error.

### PR DESCRIPTION
**Pull request by jonas054**
Here's something I've been fiddling with for a while. I think it's necessary to split the checking of multi-line binary operations and method calls into two cops. It gets messy otherwise.

I chose to support alignment such as

```ruby
users = User.where(email: 'user@example.com')
            .update_all(user_parameters)
```
within the `aligned` style, rather than introducing a new style. People who prefer to align all methods with the receiver can just add a line break:
```ruby
users = User
        .where(email: 'user@example.com')
        .update_all(user_parameters)
```

### CODE REVIEW

**Comment by jonas054** [comment]
Is this change difficult to review? Did I take the wrong approach in incorporating the new behavior into the existing `aligned` style? Is there anything I can do to make is easier to merge it (apart from rebasing right now)?

**Comment by bbatsov** [comment]
I've just been super busy. Sorry about this.

File: lib/rubocop/cop/mixin/multiline_expression_indentation.rb [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# encoding: utf-8
+
+require 'pp'
+
+module RuboCop
+  module Cop
+    # Common functionality for checking multiline method calls and binary
+    # operations.
```
Comment (bbatsov) [inline lib/rubocop/cop/mixin/multiline_expression_indentation.rb]: Here we should have some code examples.

Comment (bbatsov) [inline lib/rubocop/cop/mixin/multiline_expression_indentation.rb]: Ops, too early in the morning - this is just some mixin.

**Comment by bbatsov** [comment]
Looks good to me. Just rebase it and you can merge it.

**Comment by jonas054** [comment]
Rebased. Finally. :smile:

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**